### PR TITLE
[basicprofiles] Add $DELTA_PERCENT function to State Filter profile

### PR DESCRIPTION
**Pull request by jimtng**
Support `$DELTA_PERCENT > 5` or simply `> 5%`

### CODE REVIEW

File: bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java [hunk 1]
```diff
@@ -664,5 +674,25 @@ public String toString() {
...
+            State stateDelta = calculateDelta();
+            if (stateDelta == null) {
+                return null;
+            }
+
+            BigDecimal bdDelta;
+            BigDecimal bdBase;
+            if (deltaState instanceof QuantityType deltaStateQuantity) {
+                // Assume that delta and base are in the same unit
+                bdDelta = ((QuantityType) stateDelta).toBigDecimal();
```
Comment (lsiepel) [inline bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java:688]: This is somewhat confusing and possibly wrong.
1. The method starts with assigning `stateDelta` but deos not seem to be used later on.
2. Iin line 686 `deltaState` (reverse name?!) is used with instanceof
3. The var `deltaStateQuantity` created with the instanceof is not used. 
4. Within the instanceof block another case is done. `((QuantityType) stateDelta)`

Comment (jimtng) [reply bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java]: > 3\. The var `deltaStateQuantity` created with the instanceof is not used.

Thanks for spotting this. That part was written months ago and I forgot to use it. 

I have also renamed stateDelta and deltaState so it makes more sense and avoid ambiguity for the reader. No changes in logic.

Comment (lsiepel) [reply bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java]: Thanks, better now.

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

**Pull request merged**